spec/services/discourse_service_spec.rb

```diff
@@ -152,4 +152,82 @@
       WebMock.disable_net_connect!
     end
   end
+
+  describe ".create_email_token" do
+    it "creates a email token" do
+      user = FactoryBot.create(:user)
+
+      discourse = DiscourseService.new(user)
+
+      response = {
+        "success" => "OK",
+        "user_found" => true,
+        "email_token" => "15fa6c1c626cb97ccf18e5abd537260e"
+      }
+
+      stub_discourse_request(:post, "u/email-token.json")
+        .to_return(status: 200, body: response.to_json, headers: {"Content-Type": "application/json"})
+
+      response = discourse.create_email_token
+
+      expect(response["success"]).to eq("OK")
+      expect(response["user_found"]).to eq(true)
+      expect(response["email_token"]).to eq("15fa6c1c626cb97ccf18e5abd537260e")
+    end
+
+    it "returns user_found false when no user is found" do
+      user = FactoryBot.create(:user, email: "notfound@example.com")
+      discourse = DiscourseService.new(user)
+
+      response = {
+        "success" => "OK",
+        "user_found" => false,
+        "email_token" => nil
+      }
+
+      stub_discourse_request(:post, "u/email-token.json", {login: user.email})
+        .to_return(status: 200, body: response.to_json, headers: {"Content-Type": "application/json"})
+
+      response = discourse.create_email_token
+
+      expect(response["success"]).to eq("OK")
+      expect(response["user_found"]).to eq(false)
+      expect(response["email_token"]).to be_falsy
+    end
+
+    it "creates an email token" do
+      user = FactoryBot.create(:user, email: "orlando@debtcollective.org")
+
+      response = {
+        "success" => "OK",
+        "user_found" => true,
+        "email_token" => "29fa89dd95e1f5008fbad2b3fca9011e"
+      }
+
+      stub_discourse_request(:post, "u/email-token.json", {login: user.email})
+        .to_return(status: 200, body: response.to_json, headers: {"Content-Type": "application/json"})
+
+      discourse = DiscourseService.new(user)
+      response = discourse.create_email_token
+
+      expect(response["success"]).to eq("OK")
+      expect(response["user_found"]).to eq(true)
+      expect(response["email_token"]).to be_truthy
+    end
+
+    xit "creates a email token against the API" do
```

Comment:
It is ok to skip this one? Seems to be the most important one

Reply:
This is to test against the Discourse server. I was using this test to develop the implementation. I'm leaving this one disabled because we don't have a configured discourse instance to test against it. I'm mocking the result in the other tests and we have tests for this endpoint here https://github.com/debtcollective/discourse-debtcollective-sso/blob/master/spec/requests/users_controller_spec.rb#L49